pkg/api/pod/util.go

```diff
@@ -23,6 +23,18 @@ import (
 	"k8s.io/kubernetes/pkg/util/intstr"
 )
 
+const (
+	// The annotation value is a string specifying the hostname to be used for the pod e.g 'my-webserver-1'
+	PodHostnameAnnotation = "pod.beta.kubernetes.io/hostname"
+
+	// The annotation value is a string specifying the subdomain e.g. "my-web-service"
+	// If specified, on the the pod itself, "<hostname>.my-web-service.<namespace>.svc.<cluster domain>" would resolve to
+	// the pod's IP.
+	// If there is a headless service named "my-web-service" in the same namespace as the pod, then,
+	// <hostname>.my-web-service.<namespace>.svc.<cluster domain>" would be resolved by the cluster DNS Server.
+	PodSubdomainAnnotation = "pod.beta.kubernetes.io/subdomain"
```

Comment:
@liggitt and I are still confused on this.

>>Also, the pod must be an endpoint for the headless service...

Why use a headless service?  Why not a service that matches a single pod by label?

>>This allows the pod to declare that it expects an FQDN and that the form of the FQDN will match the names that we add to DNS for service endpoints.

This confuses us.  Can you explain why this is needed?  It's less obvious to me than the pod hostname.